nncf/experimental/onnx/graph/model_transformer.py

```diff
@@ -11,118 +11,102 @@
-            var_out = []
-            for out in outputs:
-                # shape should be None; if you place not None, some models will have inference problems (e.g. Mask RCNN)
-                type_proto = onnx.helper.make_tensor_type_proto(onnx_graph.get_edge_dtype(out),
-                                                                shape=None)
-                value_info = onnx.helper.make_value_info(name=out, type_proto=type_proto)
-                var_out.append(value_info)
-
-            graph = onnx.helper.make_graph(nodes=model.graph.node,
-                                           name=model.graph.name,
-                                           inputs=model.graph.input,
-                                           outputs=var_out,
-                                           initializer=model.graph.initializer,
-                                           value_info=model.graph.value_info)
-            onnx_model = onnx.helper.make_model(graph,
-                                                ir_version=model.ir_version,
-                                                producer_name=model.producer_name,
-                                                producer_version=model.producer_version,
-                                                domain=model.domain,
-                                                model_version=model.model_version,
-                                                doc_string=model.doc_string)
-            if len(model.metadata_props) > 0:
-                values = {p.key: p.value for p in model.metadata_props}
-                onnx.helper.set_model_props(onnx_model, values)
-
-            if len(onnx_model.graph.input) != len(model.graph.input):
-                raise RuntimeError("Input mismatch {} != {}".format(
-                    len(onnx_model.input), len(model.input)))
-            # fix opset import
-            del onnx_model.opset_import[:]
-            for oimp in model.opset_import:
-                op_set = onnx_model.opset_import.add()
-                op_set.domain = oimp.domain
-                op_set.version = oimp.version
-            return onnx_model
-
-        onnx_graph = ONNXGraph(self.transformed_model)
-        nncf_graph = GraphConverter.create_nncf_graph(self.transformed_model)
-        model_outputs = [output.name for output in onnx_graph.get_model_outputs()]
+        self._model = deepcopy(model)
+        self._transformation_layout = ONNXTransformationLayout
+        self._output_insertion_command = ONNXOutputInsertionCommand
+        self._quantizer_insertion_command = ONNXQuantizerInsertionCommand
+        self._callbacks_by_commands = {
+            self._output_insertion_command: self._apply_output_insertion_transformations,
+            self._quantizer_insertion_command: self._apply_quantizer_insertion_transformations,
+        }
+
+    def _apply_output_insertion_transformations(self, transformations: List[ONNXOutputInsertionCommand]):
```

Comment:
This method could be in `StaticModelTransformerBase`. Isn't it?

Reply:
It can, yes. But currently, adding this method into the Base class will entail adding other required methods: `_get_backend_graph`,  `_get_regular_model_outputs`, and others. I'm not sure that these methods can be abstracted now.
This is why this method was added to `StaticModelTransformerBase` as an abstract.